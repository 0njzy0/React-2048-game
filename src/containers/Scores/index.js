import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styles from './scores.scss';

function Title({ score, bestScore }) {
  const { index, text, best } = styles;
  return (
    <div className={index}>
      <h2 className={styles.score}>
        <p className={text}>SCORE</p>
        <p>
          {score}
        </p>
      </h2>
      <h2 className={`${styles.score} ${best}`}>
        <p className={text}>BEST</p>
        <p>
          { bestScore }
        </p>
      </h2>
    </div>
  );
}

Title.propTypes = {
  score: PropTypes.number.isRequired,
  bestScore: PropTypes.number.isRequired,
};

const mapStateToProps = state => ({
  score: state.board.score,
  bestScore: state.board.bestScore,
});

export default connect(mapStateToProps)(Title);
