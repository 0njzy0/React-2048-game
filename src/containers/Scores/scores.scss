.index {
  display: flex;
  justify-content: center;

  .score {
    background: transparent;
    border-radius: 0;
    border-width: 0;
    color: #fff;
    font-size: 18px;
    font-weight: 500;
    height: 40px;
    margin: 0;
    padding: 12px 25px;
    position: relative;
    z-index: 0;
  }

  .text {
    color: #fafafa;
    font-size: 12px;
    margin-bottom: 6px;
  }

  .score::before {
    // background: #fafafa;
    background: rgba(29, 137, 255, 0.9);
    content: '';
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    transform: skewX(20deg);
    width: 100%;
    z-index: -1;
  }

  .score::after {
    background: rgba(250, 250, 250, 0.3);
    content: '';
    height: 100%;
    left: 0;
    opacity: 0;
    position: absolute;
    top: 0;
    transform: skewX(20deg);
    width: 0;
    z-index: -1;
  }

  .best {
    margin-left: 15px;
  }
}
