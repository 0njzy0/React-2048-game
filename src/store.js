import { createStore } from 'redux';
import rootReducer from './reducers';

/* eslint-disable no-underscore-dangle */
const store = createStore(rootReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ &&
  window.__REDUX_DEVTOOLS_EXTENSION__());

let preBoard = null;
store.subscribe(() => {
  const { board } = store.getState();
  const prev = JSON.stringify(preBoard);
  const curr = JSON.stringify(board);
  if (prev !== curr) {
    localStorage.setItem('board', curr);
    preBoard = board;
  }
});

export default store;
