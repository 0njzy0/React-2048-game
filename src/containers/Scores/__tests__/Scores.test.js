import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../store';
import Scores from '..';

describe('<Scores />', () => {
  it('component render', () => {
    const scores = renderer.create(
      <Provider store={store}>
        <Scores score={1234} bestScore={4321} />
      </Provider>,
    ).toJSON();
    expect(scores).toMatchSnapshot();
  });
});
