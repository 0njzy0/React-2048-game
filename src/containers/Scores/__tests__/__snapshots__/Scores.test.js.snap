// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Scores /> component render 1`] = `
<div
  className="index"
>
  <h2
    className="score"
  >
    <p
      className="text"
    >
      SCORE
    </p>
    <p>
      0
    </p>
  </h2>
  <h2
    className="score best"
  >
    <p
      className="text"
    >
      BEST
    </p>
    <p>
      0
    </p>
  </h2>
</div>
`;
